File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.py
class Solution:
  def longestSubarray(self, nums: List[int]) -> int:
    maxNum = max(nums)
    ans, contiguous = 0, 0
    for num in nums:
      if num == maxNum:
        contiguous += 1
        if contiguous > ans:
          ans = contiguous
      else:
        contiguous = 0
    return ans
        

File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.php
class Solution {
  function longestSubarray($nums) {
    $max = max($nums);
    $ans = 0;
    $contiguous = 0;
    foreach ($nums as $num) {
      if ($num === $max) {
        if (++$contiguous > $ans) $ans = $contiguous;
      } else {
        $contiguous = 0;
      }
    }   
    return $ans;
  }
}